extension/webviews/components/SearchViewer.svelte

```diff
@@ -0,0 +1,65 @@
+<script lang="ts">
+  import { onMount } from 'svelte';
+  import type { Thread as ThreadType } from '../types';
+  import { Page } from '../enums';
+  import { currentProject, page } from '../state/store';
+  import Thread from './thread/Thread.svelte';
+  import Button from './Button.svelte';
+
+  let searchQuery: string = '';
+  let threads: Array<ThreadType> = [];
+
+  const handleSearchQuery = (event: any) => {
+    if (event.key === 'Enter') {
+      searchQuery = event.target.value;
+      if ($currentProject) {
+        tsvscode.postMessage({
+          type: 'searchThreads',
+          value: { searchText: searchQuery, projectId: $currentProject.id },
+        });
+      }
+    }
+  };
+
+  const handleCancelButtonClick = () => {
+    searchQuery = '';
+    $page = Page.ThreadsViewer;
+  };
+
+  const messageEventListener = async (event: any) => {
+    const message = event.data;
+    switch (message.type) {
+      case 'searchThreads':
+        threads = message.value;
+        break;
+    }
+  };
+
+  onMount(async () => {
+    if ($currentProject === null) {
+      $page = Page.InitializeProject;
+    }
```

Comment:
Is this necessary?

Reply:
yup. coz we don't wanna show the searchButton/the ViewerTopBar  if no project is selected.